app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt

```diff
@@ -208,7 +214,13 @@ class StateFragmentPresenter @Inject constructor(
   fun handleKeyboardAction() {
     hideKeyboard()
     if (viewModel.getCanSubmitAnswer().get() == true) {
-      handleSubmitAnswer(viewModel.getPendingAnswer(recyclerViewAssembler))
+      var answer = UserAnswer.getDefaultInstance()
+      if (recyclerViewAssembler.getPendingAnswerHandler(viewModel.itemList) != null) {
+        answer = viewModel.getPendingAnswerWithoutError(
+          recyclerViewAssembler.getPendingAnswerHandler(viewModel.itemList)!!
+        )
+      }
+      handleSubmitAnswer(answer)
```

Comment:
Use a helper to avoid the code duplication here.

Reply:
Okay!